autoPyTorch/data/time_series_forecasting_validator.py

```diff
@@ -0,0 +1,355 @@
+                x_transformed, _ = self._transform_X(X, None)
+                return x_transformed, None, X.index.value_counts(sort=False).values
+
+            else:
+                raise NotImplementedError
+        else:
+            if y is None:
+                raise ValueError("Targets must be given!")
+
+            if isinstance(y, List):
+                num_sequences = len(y)
+                sequence_lengths = [0] * num_sequences
+                if not self._is_uni_variant:
+                    if X is None:
+                        raise ValueError("Multi Variant dataset requires X as input!")
+                    assert len(X) == len(
+                        y
+                    ), "Length of features must equal to length of targets!"
+                if self.series_idx is not None and X is None:
+                    raise ValueError("X must be given as series_idx!")
+
+                for seq_idx in range(num_sequences):
+                    sequence_lengths[seq_idx] = len(y[seq_idx])
+                npa_sequence_lengths = np.asarray(sequence_lengths)
+
+                y_stacked = self.join_series(y)
+
+                x_transformed, series_number = self._transform_X(
+                    X, npa_sequence_lengths
+                )
+                y_transformed = self.target_validator.transform(
+                    y_stacked, index=series_number
+                )
+
+                if self._is_uni_variant:
+                    return None, y_transformed, npa_sequence_lengths
+
+                return x_transformed, y_transformed, npa_sequence_lengths
+            elif isinstance(y, (pd.DataFrame, pd.Series)):
+                if self.series_idx is not None:
+                    if isinstance(y, pd.Series):
+                        y_columns = [y.name]
+                    else:
+                        if isinstance(y.columns, pd.RangeIndex):
+                            y_columns = [f"target_{i}" for i in y.columns]
+                            y.columns = y_columns
+                        y_columns = y.columns
+                    xy = pd.concat([X, y], axis=1)
+                    xy.sort_values(self.feature_validator.series_idx, inplace=True)
+
+                    y = xy[y_columns]
+                    X = xy.drop(y_columns, axis=1)
+                    del xy
+
+                x_transformed, series_number = self._transform_X(X, None)
+
+                if self._is_uni_variant:
+                    y_transformed = self.target_validator.transform(
+                        y, series_number
+                    )
+                    return (
+                        None,
+                        y_transformed,
+                        y_transformed.index.value_counts(sort=False).values,
+                    )
+
+                y_transformed = self.target_validator.transform(
+                    y, x_transformed.index
+                )
+                return (
```

Comment:
Hey, I just noticed that you are not compressing the dataset. Do you think we can also combine it with time series tasks? For ref, you can see [here](https://github.com/dengdifan/Auto-PyTorch/blob/c1de20f943af811a134a2a4780c5499546af45f2/autoPyTorch/data/tabular_validator.py#L119). You can simply call this line at the end of the transform function in this validator.

Reply:
Hmmm... I applied a different way to store the data. The current judgment might not be valid for my settings. Additionally, I need a specific type of subsampling approach. But I will leave it as TODO

Reply:
okay no problem, could you also raise an issue for this?